### PR TITLE
Ticket/57701: Disable prime cache by default in wp_get_object_terms.

### PR DESCRIPTION
**Pull request by rutviksavsani**
With this PR the term meta will not be primed by default. It sets `update_term_meta_cache` value to false as default.

Trac ticket: https://core.trac.wordpress.org/ticket/57701

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/term/wpGetObjectTerms.php [hunk 1]
```diff
@@ -614,6 +614,7 @@ public function test_parent_0() {
 
 	/**
 	 * @ticket 10142
+	 * @ticket 57701: Updated test to reflect that wp_get_object_terms does not prime the cache.
 	 */
 	public function test_termmeta_cache_should_be_lazy_loaded_by_default() {
```
Comment (spacedmonkey) [inline tests/phpunit/tests/term/wpGetObjectTerms.php]: Let's change the name of this test as well.

File: tests/phpunit/tests/term/wpGetObjectTerms.php [hunk 2]
```diff
@@ -614,6 +614,7 @@ public function test_parent_0() {
 
 	/**
 	 * @ticket 10142
+	 * @ticket 57701: Updated test to reflect that wp_get_object_terms does not prime the cache.
```
Comment (spacedmonkey) [inline tests/phpunit/tests/term/wpGetObjectTerms.php]: Comment not need here.

File: tests/phpunit/tests/term/getTheTerms.php [hunk 3]
```diff
@@ -195,8 +195,9 @@ public function test_count_should_not_be_improperly_cached() {
 
 	/**
 	 * @ticket 36814
+	 * @ticket 57701: Updated test to reflect that wp_get_object_terms does not prime the cache.
```
Comment (spacedmonkey) [inline tests/phpunit/tests/term/getTheTerms.php]: Comment not need here.

File: tests/phpunit/tests/term/getTheTerms.php [hunk 4]
```diff
@@ -213,9 +214,8 @@ public function test_uncached_terms_should_be_primed_with_a_single_query() {
 
 		$this->assertSameSets( $terms, wp_list_pluck( $found, 'term_id' ) );
 
-		$num_queries++;
-		$this->assertSame( $num_queries, get_num_queries() );
-
+		// Two extra queries are expected as the cache is not primed and hence terms need to be queried.
+		$this->assertSame( $num_queries + 1, get_num_queries() );
```
Comment (spacedmonkey) [inline tests/phpunit/tests/term/getTheTerms.php]:
```suggestion
		$this->assertSame( 1, get_num_queries() - $num_queries );
```

File: tests/phpunit/tests/term/wpGetObjectTerms.php [hunk 5]
```diff
@@ -633,7 +634,8 @@ public function test_termmeta_cache_should_be_lazy_loaded_by_default() {
 			$this->assertSame( 'bar', get_term_meta( $t, 'foo', true ) );
 		}
 
-		$this->assertSame( $num_queries + 1, get_num_queries() );
+		// Here we had extra queries as the term meta cache was not primed by default.
+		$this->assertSame( $num_queries + 3, get_num_queries() );
```
Comment (spacedmonkey) [inline tests/phpunit/tests/term/wpGetObjectTerms.php]:
```suggestion
		$this->assertSame( 3, get_num_queries() - $num_queries );
```

File: tests/phpunit/tests/term/wpGetObjectTerms.php (hunk 1 above)
Comment (rutviksavsani) [reply tests/phpunit/tests/term/wpGetObjectTerms.php]: Anything specific you have in mind here as a replacement name?

File: tests/phpunit/tests/term/wpGetObjectTerms.php (hunk 2 above)
Comment (rutviksavsani) [reply tests/phpunit/tests/term/wpGetObjectTerms.php]: Should I remove the entire comment?

Comment (spacedmonkey) [reply tests/phpunit/tests/term/wpGetObjectTerms.php]: leave `@ticket 57701`

File: tests/phpunit/tests/term/wpGetObjectTerms.php (hunk 1 above)
Comment (spacedmonkey) [reply tests/phpunit/tests/term/wpGetObjectTerms.php]: `test_termmeta_cache_not_should_be_lazy_loaded_by_default`

Comment (rutviksavsani) [reply tests/phpunit/tests/term/wpGetObjectTerms.php]: done

File: tests/phpunit/tests/term/wpGetObjectTerms.php (hunk 2 above)
Comment (rutviksavsani) [reply tests/phpunit/tests/term/wpGetObjectTerms.php]: done

**Comment by spacedmonkey** [comment]
Committed in https://core.trac.wordpress.org/changeset/55759

**Pull request closed without merging**